be/src/storage/schema_change.cpp

```diff
@@ -1277,6 +1283,15 @@ bool SchemaChangeWithSorting::process(RowsetReaderSharedPtr rowset_reader, Rowse
             LOG(WARNING) << "failed to allocate RowBlock.";
             result = false;
             goto SORTING_PROCESS_ERR;
+        } else {
+            // check allocator has enougth memory to do row-block sort
+            if (!_row_block_allocator->can_sort(ref_row_block->row_block_info().row_num,
+                                                row_block_sorter.allocated_rows())) {
+                if (new_row_block != nullptr) {
```

Comment:
when memory is not enough, if new_row_block is set to null.
Is the final result still the right one?